app/Http/Controllers/CommentController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\StoreCommentRequest;
+use App\Http\Requests\UpdateCommentRequest;
+use App\Models\Comment;
+use App\Models\Post;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\Request;
+use Illuminate\Http\Exceptions\HttpResponseException;
+
+
+
+
+
+class CommentController extends Controller
+{
+
+    public function findOrFail($id): Post|JsonResponse
+    {
+        $post = Post::find($id);
+        
+        if (!$post) {
+            throw new HttpResponseException(
+            response()->json([
+                'status' => false,
+                'message' => 'Post Not Found',
+            ], 404)
+        );
+        }
+        
+        return $post;
+    }
+
+    /**
+     * Display a listing of the resource.
+     */
+    public function index($id)
+    {
+        $post = $this->findOrFail($id);
+
+        $comments = $post->comments()->with('replies')->cursorPaginate(10);
+        return [ 
+            'status'=>true,
+            'data' => ['comments'=>$comments]
+        ];
+    }
```

Comment:
1. Naming the url param `$id` will be confusing. You can use `$postId`
2. Use route model binding.
```php
public function index(Post $post)
```